### ISSUE TITLE
Modify authentication to DB to use AD account instead of cactiuser

### ISSUE THREAD

**Issue by andermat8** [Cacti/cacti#1479]
Copying text from a previous issue as this is separate from #1457 
andermat8 commented 3 hours ago
Also, this is probably a new item / request but has there been any consideration in changing how clear-text passwords are stored in the config.php and spine.conf. It is just a matter of time before I get hammered on an audit for this.

 @netniV
Contributor
netniV commented 3 hours ago
Don't take this as gospel but from my view point, that is a standard Linux configuration. Most passwords are stored there in plain text but have access only to the database of the software in question. Even if we encrypted the password, there would be a function within the PHP code to decrypt this so I can't see that really adding much protection except to the most basic script kiddie.

Whatever user you use for Cacti should only EVER have access to the cacti database.

 @andermat8
  
andermat8 commented 2 hours ago
Yup, I understand but I don't think I'll be able to convince our Security team.

 @netniV
Contributor
netniV commented 2 hours ago
How are they dealing with other such packages?

 @andermat8
  
andermat8 commented an hour ago
I am running Cacti on a Windows server (long story as to why) but I know for our other applications DB connections are encrypted. There are no applications that I'm aware of in our environment, aside from Cacti, that stores passwords in plain-text.

 @netniV
Contributor
netniV commented an hour ago
My bad, it was mentioned above about that. So the main reason that Cacti doesn't have this by default is that you have to customise the MySQL installation to include the Windows Authentication plugin:
https://dev.mysql.com/doc/refman/5.5/en/windows-pluggable-authentication.html

Most MySQL installations will not have that. I guess we could add an option to not supply the password should it be on a windows platform though I'm not sure how we would know if that plugin was set. The only thing we could do is try to connect.

 @andermat8
  
andermat8 commented 31 minutes ago
Yes, it would be great if we could use an AD service account to authenticate to the DB instead.

 @netniV
Contributor
netniV commented 28 minutes ago
I would imagine that if we left off the requirement for a password, that would be as far as we would go. I suspect it would be considered the DB admin's responsibility for installation and configuration of the extra plugin to MySQL.

Such a change I would imagine would be a 1.2 change at this point since it's more a feature request than bug per sae. So, could you open a separate issue for that and then I will update with the proposal. Once that is done, we can then remove the authentication comments from this issue, to keep it on track about the XSS stuff.

**Comment by netniV** [comment]
OK, so I have tried and failed to find any decent documentation on how to do this.  Annoying the MySQL site goes around in loops and ends up at the same page which basically says how to setup Windows Authentication at the database level.  What seems to be missing these days are any documentation on how PHP or C++ can utilise the Windows authentication to perform the connection.

Since connections can be made using SSL in 1.2 (which is going to hit beta soon), then I'm going to close this as the best we can do at this time.

Should they ever properly document how to perform the integration, we can review this.
